### PR TITLE
update pwr_g0 from stm32g0b1

### PR DESCRIPTION
**Pull request by stillinbeta**

### CODE REVIEW

File: data/registers/pwr_g0.yaml [hunk 1]
```diff
@@ -33,20 +33,54 @@ block/PWR:
...
+      description: Power Port A pull-up control register
       byte_offset: 32
-      fieldset: PCR
-    - name: PDCR
-      description: Power Port pull-down control register
-      array:
-        len: 6
-        stride: 8
+      fieldset: PUCR
+    - name: PDCRA
```
Comment (stillinbeta) [inline data/registers/pwr_g0.yaml:40]: is there a way to make these an array but have the letters still show up?

Comment (Dirbaio) [reply data/registers/pwr_g0.yaml:40]: Not really possible, array indices are only integers. 

There's one hack which is to keep both the PUCR array reg and the unrolled PUCRx regs. I wouldn't do it though, unrolled registers bloat the PAC and make it compile slower. The convention we're using is to arrayify everything even if it means losing letters.

**Comment by stillinbeta** [comment]
The reason I don't want to just do numbers here is that PU/DCR F is a
special case, because it only has 14 registers. So it'd look weird if we
did [0,1,2,3,4,5] F

**Comment by Dirbaio** [comment]
Make F have 16 bits too. Consistency is worth the slight loss of precision. The main GPIO module already does this. 

If we don't, it becomes impossible to write a sane HAL, because it causes different ports to have different Rust types, which needs horrible macro spaghetti abominations at the HAL layer to workaround.

**Pull request closed without merging**